applications/ecallmgr/src/ecallmgr_discovery.erl

```diff
@@ -0,0 +1,177 @@
+%%%-----------------------------------------------------------------------------
+%%% @copyright (C) 2012-2018, 2600Hz
+%%% @doc
+%%% @end
+%%%-----------------------------------------------------------------------------
+-module(ecallmgr_discovery).
+-behaviour(gen_server).
+
+-export([start_link/0]).
+
+-export([init/1
+        ,handle_call/3
+        ,handle_cast/2
+        ,handle_info/2
+        ,terminate/2
+        ,code_change/3
+        ]).
+
+-include("ecallmgr.hrl").
+
+-define(SERVER, ?MODULE).
+
+-type state() :: map().
+
+%%%=============================================================================
+%%% API
+%%%=============================================================================
+
+%%------------------------------------------------------------------------------
+%% @doc Starts the server
+%% @end
+%%------------------------------------------------------------------------------
+-spec start_link() -> kz_types:startlink_ret().
+start_link() ->
+    gen_server:start_link({'local', ?SERVER}, ?MODULE, [], []).
+
+%%%=============================================================================
+%%% gen_server callbacks
+%%%=============================================================================
+
+%%------------------------------------------------------------------------------
+%% @doc Initializes the server
+%%
+%% @end
+%%------------------------------------------------------------------------------
+-spec init(list()) -> {'ok', state(), pos_integer()}.
+init([]) -> {'ok', #{startup => kz_time:now_s()}, ?MILLISECONDS_IN_SECOND}.
```

Comment:
Is there a reason not to just track the `startup` directly? Do you have plans for more state to be tracked here?

Reply:
not sure i understand your comment, but as always, feel free to improve it.
i need to keep the initial `startup` time to be used in `next_timeout(kz_time:elapsed_s(Startup))` so that the timer will increase (more often to less often checks)

Reply:
Sorry, let me restate (heh). State is a map here. But looking through the code, I only ever see the map unpacked to get the `startup` attribute. So why not just return `{'ok', kz_time:now_s(), ?MILLISECONDS_IN_SECOND}` and use it directly in the other functions?

Reply:
ah, right. can do that change, thanks